Kubernetes/legos/k8s_get_oomkilled_pods/k8s_get_oomkilled_pods.py

```diff
@@ -44,23 +45,37 @@ def k8s_get_oomkilled_pods(handle, namespace: str = "") -> Tuple:
     # Check whether a namespace is provided, if not fetch all namespaces
     try:
         if namespace:
-            pods = v1.list_namespaced_pod(namespace).items
+            response = v1.list_namespaced_pod(namespace)
         else:
-            pods = v1.list_pod_for_all_namespaces().items
+            response = v1.list_pod_for_all_namespaces()
+        
+        if response is None or not hasattr(response, 'items'):
+            raise ApiException("Unexpected response from the Kubernetes API. 'items' not found in the response.")
+
+        pods = response.items
+
     except ApiException as e:
         raise e
 
+    # Check if pods is None or not
+    if pods is None:
+        raise ApiException("No pods returned from the Kubernetes API.")
+
     for pod in pods:
         pod_name = pod.metadata.name
         namespace = pod.metadata.namespace
-    # Check each pod for OOMKilled state
-        for container_status in pod.status.container_statuses:
+
+        # Ensure container_statuses is not None before iterating
+        container_statuses = pod.status.container_statuses
+        if container_statuses is None:
+            continue
```

Comment:
what is the case in which the `container_statuses` comes out to be `None`?

Reply:
If the pod is just created, is evicted from a node, or when there's a problem with the Kubelet on the node where the pod is scheduled, and it isn't reporting status to the API server then container_statuses can be None. This case just ensures no error is thrown in these cases. (which is why it was broken in the first place)